From dc2f2f468fc404aae3e2169ec6a16b0da366a1e0 Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Tue, 25 Nov 2014 21:05:03 -0600
Subject: [PATCH 23/26] IssuingDistributionPoint constructor fix (bc1.47)

The IssuingDistributionPoint(ASN1Sequence) constuctor was gone, so
instead I used the getInstance() factory method, as recommended by the
porting guide.

I also fixed a typo in the name of a private function (Issuin ->
Issuing).
---
 src/org/glite/security/util/FileCRLChecker.java | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/org/glite/security/util/FileCRLChecker.java b/src/org/glite/security/util/FileCRLChecker.java
index 9996a2b..747403c 100644
--- a/src/org/glite/security/util/FileCRLChecker.java
+++ b/src/org/glite/security/util/FileCRLChecker.java
@@ -276,7 +276,7 @@ public class FileCRLChecker extends RevocationChecker {
 			throw new CertificateException("Usupported critical extension in CRL: DeltaCRLIndicator");
 		}
 		if (oid.equals(X509Extensions.IssuingDistributionPoint.toString())) {
-			checkIssuinDistributionPoint();
+			checkIssuingDistributionPoint();
 			return;
 		}
 
@@ -291,7 +291,7 @@ public class FileCRLChecker extends RevocationChecker {
 	 * @throws IOException
 	 *             thrown in case the extension parsing fails.
 	 */
-	private void checkIssuinDistributionPoint() throws CertificateException, IOException {
+	private void checkIssuingDistributionPoint() throws CertificateException, IOException {
 		byte extensionBytes[] = m_crl.getExtensionValue(X509Extensions.IssuingDistributionPoint.toString());
 
 		ASN1Primitive object = ASN1Primitive.fromByteArray(extensionBytes);
@@ -305,7 +305,7 @@ public class FileCRLChecker extends RevocationChecker {
 			throw new CertificateException("Invalid data in IssuingDistributionPoint extension, not ASN1Sequence");
 		}
 
-		IssuingDistributionPoint issuingDistributionPoint = new IssuingDistributionPoint((ASN1Sequence) object);
+		IssuingDistributionPoint issuingDistributionPoint = IssuingDistributionPoint.getInstance(object);
 
 		if (issuingDistributionPoint.onlyContainsAttributeCerts()) {
 			throw new CertificateException("CRL only contains attribute certs, not useful for authentication.");
-- 
2.0.1

